study1/week3/KeysAndRooms.java

```diff
@@ -0,0 +1,38 @@
+class Solution {
+    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
+        int n = rooms.size();
+        boolean[] isVisited = new boolean[n];
+
+        dfs(0, isVisited, rooms);
+
+        for (int i = 0; i < n; i++) {
```

Comment:
You can make this loop into a single statement by 
1. Changing the isVisited variable into an `ArrayList`
2. chagrining line 8 to 13 intothe following

```java
return !isVisited.stream().anyMatch(visit -> !visit);
```

or you can also count the number of visited nodes and do `return numVisited == rooms.size();`